app/protocol/flows/ibd/ibd.go

```diff
@@ -64,7 +99,104 @@ func (flow *handleIBDFlow) runIBD() error {
 
 	log.Debugf("Found highest shared chain block %s with peer %s", highestSharedBlockHash, flow.peer)
 
-	return flow.downloadBlocks(highestSharedBlockHash, peerSelectedTipHash)
+	return flow.downloadHeaders(highestSharedBlockHash, peerSelectedTipHash)
+}
+
+func (flow *handleIBDFlow) syncMissingBlockBodies() error {
+	hashes, err := flow.Domain().Consensus().GetMissingBlockBodyHashes(flow.peer.SelectedTipHash())
```

Comment:
Is there any chance we are going to have a billion hashes here, that we need to store for some time in memory, which will kill us?
I assume answer is "premature optimization", but please give that a thought as well.